app/controllers/home_controller.rb

```diff
@@ -2,6 +2,7 @@ class HomeController < ApplicationController
   include Prerenderable
 
   skip_before_action :authenticate_user!, only: %i[index upgrade_browser]
+  render_flash_messages_via_js(only: %i[index])
```

Comment:
We might want to do this for some other controller actions, in the future, but for now the real concern is for home#index because that is the fallback/default location for redirects after an error of some kind.